### PR TITLE
Feature/auth/46 firebase

### PR DESCRIPTION
**Pull request by Isaac-D-Dawson**
Added Firebase-based sign-up/log-out handler.
Link:
https://trello.com/c/eRsUQNrf/58-add-checking-for-email-and-password

### CODE REVIEW

File: src/pages/SignUp/SignUp.jsx [hunk 1]
```diff
@@ -3,7 +3,92 @@ import Button from "../../components/Button/Button";
...
+const validateInputs = () => {
+  let result = true;
+  setEmailError("");
+  setEmailSuccess("");
+  setPassword1Error("");
+  setPassword1Success("");
+  setPassword2Error("");
+  setPassword2Success("");
+  if (!emailRegex.test(registerEmail)) {
+    setEmailError("Error - Your email must contain an @ symbol");
```
Comment (LewisLovett) [inline src/pages/SignUp/SignUp.jsx:42]: Change error messge to 'Please enter a valid email'

File: src/pages/SignUp/SignUp.jsx [hunk 2]
```diff
@@ -3,7 +3,92 @@ import Button from "../../components/Button/Button";
...
+    setEmailSuccess("Successful email entered");
+  }
+  if (confirmPassword != registerPassword) {
+    setPassword2Error("Error - passwords do not match!");
+    result = false;
+  } else {
+    setPassword2Success("Passwords Match");
+  }
+  if (!passRegex.test(registerPassword)) {
+    setPassword1Error("Error - password must contain at least 8 characters, 1 uppercase letter, and 1 symbol");
```
Comment (LewisLovett) [inline src/pages/SignUp/SignUp.jsx:54]: The error message must mention that the password needs to contain a number

File: src/pages/SignUp/SignUp.jsx [hunk 3]
```diff
@@ -3,7 +3,92 @@ import Button from "../../components/Button/Button";
...
+    result = false;
+  }
+  if (confirmPassword === registerPassword && passRegex.test(registerPassword)) {
+    setPassword1Success("Password meets requirements.")
+    result = false;
```
Comment (LewisLovett) [inline src/pages/SignUp/SignUp.jsx:59]: Remove 'result = false' on line 59 what causes function to return false even if the inputs are correct

File: src/pages/SignUp/SignUp.jsx [hunk 4]
```diff
@@ -57,7 +146,7 @@ const SignUp = ({ login }) => {
         </div>
 
         <div className="signUp__continue">
-          <Button buttonText="Continue >" />
+          <Button buttonText="Continue >" handleClick={register} />
```
Comment (LewisLovett) [inline src/pages/SignUp/SignUp.jsx:149]: Once the user has successfully created and account they should be taken to the sign in page.

**Review by LewisLovett — CHANGES_REQUESTED** [review]

File: src/pages/SignUp/SignUp.jsx (hunk 4 above)
Comment (Isaac-D-Dawson) [reply src/pages/SignUp/SignUp.jsx:149]: By default when this is completed the user will be signed in automatically. Should we take them to the Dashboard instead?

**Review by LewisLovett — APPROVED** [review]

**Review by olirob93 — APPROVED** [review]
Really great work Hao & Isaac! Me and Lannie added a couple commits to get it over the line

**Review by LewisLovett — APPROVED** [review]

**Pull request merged**